servatrice/src/serversocketinterface.cpp

```diff
@@ -747,6 +749,54 @@ Response::ResponseCode ServerSocketInterface::cmdReplayDeleteMatch(const Command
 // MODERATOR FUNCTIONS.
 // May be called by admins and moderators. Permission is checked by the calling function.
 
+Response::ResponseCode ServerSocketInterface::cmdGetBanHistory(const Command_GetBanHistory &cmd, ResponseContainer &rc)
+{
+    if (!sqlInterface->checkSql())
+        return Response::RespInternalError;
+
+    QString userName = QString::fromStdString(cmd.user_name());
+
+    //lookup users ban history from database
+    QSqlQuery *query = sqlInterface->prepareQuery("select id_admin,time_from,minutes,reason,visible_reason from {prefix}_bans where user_name = :user_name");
+    query->bindValue(":user_name", userName);
+    sqlInterface->execSqlQuery(query);
+
+    if (!sqlInterface->execSqlQuery(query))
+        return Response::RespInternalError;
+
+    //build response container information from results
+    QString adminID,adminName,banTime,banLength,banReason,visibleReason;
+    Response_BanHistory *re = new Response_BanHistory;
+    while (query->next()){
+
+        adminID = query->value(0).toString();
+        adminName = "Uknown";
+        banTime = query->value(1).toString();
+        banLength = query->value(2).toString();
+        banReason = query->value(3).toString();
+        visibleReason = query->value(4).toString();
+
+        //lookup admin name based on user id from ban table
+        QSqlQuery *anamequery = sqlInterface->prepareQuery("select name from cockatrice_users where id = :user_id");
```

Comment:
You could avoid making  second query if you join the two tables in the first query:
```
SELECT A.id_admin, A.time_from, A.minutes, A.reason, A.visible_reason, B. name AS name_admin FROM {prefix}_bans A LEFT JOIN {prefix}_users B ON A.id_admin=B.id WHERE A.user_name = :user_name
```